src/fireedge/src/modules/components/Tables/VrTemplates/DatastoreSelectionDialog.js

```diff
@@ -0,0 +1,170 @@
+ * @returns {Component} - Datastore selection dialog
+ */
+export const DatastoreDialog = ({
+  open,
+  onClose,
+  datastores,
+  submitDisabled,
+  onSelect,
+}) => {
+  const theme = useTheme()
+  const classes = useMemo(() => rowStyles(theme), [theme])
+
+  const [searchTerm, setSearchTerm] = useState('')
+  const [currentPage, setCurrentPage] = useState(0)
+  const [selectedDatastore, setSelectedDatastore] = useState(null)
+  const itemsPerPage = 5
+
+  const filteredDatastores = datastores?.filter((datastore) =>
+    datastore.NAME.toLowerCase().includes(searchTerm.toLowerCase())
+  )
+
+  const paginatedDatastores = filteredDatastores.slice(
+    currentPage * itemsPerPage,
+    (currentPage + 1) * itemsPerPage
+  )
+
+  const handleSearchChange = (event) => {
+    setSearchTerm(event.target.value)
+    setCurrentPage(0)
+  }
+
+  const handleSelectDatastore = (datastore) => {
+    setSelectedDatastore(datastore)
+  }
+
+  const handleConfirmSelection = () => {
+    onSelect(selectedDatastore)
+    onClose()
+  }
+
+  const handlePrevPage = () => {
+    setCurrentPage((prevPage) => Math.max(prevPage - 1, 0))
+  }
+
+  const handleNextPage = () => {
+    setCurrentPage((prevPage) =>
+      Math.min(
+        prevPage + 1,
+        Math.ceil(filteredDatastores.length / itemsPerPage) - 1
+      )
+    )
+  }
+
+  return (
+    <Dialog open={open} onClose={onClose} maxWidth={'md'} fullWidth>
+      <DialogTitle>Select datastore for importing image</DialogTitle>
+      <DialogContent>
+        <TextField
+          label="Search by name"
+          variant="outlined"
+          fullWidth
+          margin="normal"
+          onChange={handleSearchChange}
+        />
+        <Box
+          sx={{
+            mt: 2,
+            overflow: 'auto',
+          }}
+        >
+          {paginatedDatastores.map((datastore) => (
+            <Box
+              key={datastore.ID}
+              sx={{
+                width: '100%',
+                mb: 2,
+                cursor: 'pointer',
+                border:
+                  selectedDatastore?.ID === datastore.ID
+                    ? `2px solid ${theme.palette.primary.main}`
+                    : `2px solid ${theme.palette.divider}`,
+                '&:hover': {
+                  backgroundColor: theme.palette.action.hover,
+                },
+              }}
+              onClick={() => handleSelectDatastore(datastore)}
+              className={classes.root}
+            >
+              <DatastoreCard
+                datastore={datastore}
+                rootProps={{
+                  className: classes.root,
+                  style: {
+                    width: '100%',
+                  },
+                }}
+              />
+            </Box>
+          ))}
+        </Box>
+      </DialogContent>
+      <DialogActions>
+        <Button onClick={handlePrevPage} disabled={currentPage === 0}>
+          Prev
+        </Button>
+        <Button
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

JSX element content is not internationalized. Application should support multiple languages through internationalization. Instead, utilize packages like &#39;i18next&#39; to internationalize your items.